src/plugins/numberline/models/numberline-content.ts

```diff
@@ -0,0 +1,159 @@
+    },
+    get pointsIsHoveredArr(){
+      return Array.from(self.points.values()).map((pointObj) => pointObj.isHovered);
+    },
+    get pointsIsSelectedArr(){
+      return Array.from(self.points.values()).map((pointObj) => pointObj.isSelected);
+    },
+  }))
+  .views(self =>({
+     //Pass snapshot of axisPoint models into outer/inner points to avoid D3 and MST error
+    get axisPointsSnapshot(){
+      return self.axisPoints.map((p)=> ({
+                                          isHovered: p.isHovered,
+                                          isSelected: p.isSelected,
+                                          ...getSnapshot(p) //doesn't capture the volatile properties
+                                        }));
+    },
+    get isHoveringOverPoint(){
+      return !(self.pointsIsHoveredArr.filter(Boolean).length === 0);
+    },
+    get indexOfPointHovered(){
+      return self.pointsIsHoveredArr.findIndex((isHovered) => isHovered === true);
+    }
+  }))
+  .actions(self => ({
+    setAllSelectedFalse(){
+      self.points.forEach((point)=>{
+        point.isSelected = false;
+      });
+    },
+    setAllHoversFalse(){
+      self.points.forEach((point)=>{
+        point.isHovered = false;
+      });
+    },
+    clearAllPoints(){
+      self.points.clear();
+    },
+
+    pointById(id: string){
+      return self.points.get(id);
+    },
+
+    isDraggingUseIdReplacePointCoordinates(oldPoint: PointObjectModelType, newPointCoordinates: IPointCoordinate){
+      //searches "points", removes PointObject at index that matches id
+      //replaces it with a new PointObject at index that has newPointCoordinates
+      self.axisPoints.forEach((pointObj, i) => {
+        if (pointObj.id === oldPoint.id){
+          const newPointObj = PointObjectModel.create({
+            id: oldPoint.id,
+            pointCoordinates: newPointCoordinates,
+          });
+          newPointObj.setIsHovered(oldPoint.isHovered);
+          newPointObj.setIsSelected(oldPoint.isSelected);
+          self.points.set(newPointObj.id, newPointObj);
+        }
+      });
+    },
+    replaceAllPoints(newPoints: PointObjectModelType[]){
+      self.points.replace(newPoints);
+    },
+    isMouseHoverOverPoint(mouseXPos: number, axisWidth: number ){
+      if (self.hasPoints){
+        const xScale = scaleLinear()
+          .domain([numberlineDomainMin, numberlineDomainMax])
+          .range([0, axisWidth]);
+
+        self.pointsXValuesArr.forEach((pointXValue: number, idx)=>{
+          const pointXPos = xScale(pointXValue);
+          const pointXLeftBound = pointXPos - 5;
+          const pointXRightBound = pointXPos + 5;
+          if (mouseXPos > pointXLeftBound && mouseXPos < pointXRightBound){
+            if (self.pointsIsHoveredArr.filter(Boolean).length === 0){
+              self.axisPoints[idx].isHovered = true; //only one is true
+            }
+          }
+          else{
+            self.axisPoints[idx].isHovered = false;
```

Comment:
I think it's misleading to modify state in a function that's called `is...`. I would expect this to just return a boolean.